### PR TITLE
Set up test

### PR DESCRIPTION
**Pull request by winston0410**
Set up test with jest

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const customJestConfig = {
+    setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
+    testEnvironment: 'jest-environment-jsdom',
+    coverageReporters: ['json-summary', 'text', 'lcov'],
+    collectCoverageFrom: ['src/**/*.{ts,tsx}'],
+    moduleNameMapper: {
+        "\\.(css|less|scss|sass)$": "identity-obj-proxy"
```
Comment (vadim-kirpun) [inline jest.config.js:7]: Let's remove `less` from here since we don't use it

File: jest.setup.js [hunk 2]
```diff
@@ -0,0 +1 @@
+import '@testing-library/jest-dom/extend-expect';
```
Comment (vadim-kirpun) [inline jest.setup.js:1]: As I know for the latest Jest better to shorten it to `import '@testing-library/jest-dom';`

File: package.json [hunk 3]
```diff
@@ -20,15 +24,20 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@babel/preset-env": "^7.16.11",
```
Comment (vadim-kirpun) [inline package.json:27]: Please move all `babel` packages to `devDependencies`

File: package.json [hunk 4]
```diff
@@ -20,15 +24,20 @@
...
+    "@babel/preset-react": "^7.16.7",
+    "babel-jest": "^27.4.6",
     "classnames": "^2.3.1",
-    "lbh-frontend": "^3.6.1"
+    "lbh-frontend": "^3.6.1",
+    "react-test-renderer": "^17.0.2"
   },
   "peerDependencies": {
     "lbh-frontend": "^3.6.1",
```
Comment (vadim-kirpun) [inline package.json:35]: Let's remove `lbh-frontend` from `peer` since we have it in `dependencies`

File: package.json [hunk 5]
```diff
@@ -20,15 +24,20 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@babel/preset-env": "^7.16.11",
+    "@babel/preset-react": "^7.16.7",
+    "babel-jest": "^27.4.6",
     "classnames": "^2.3.1",
-    "lbh-frontend": "^3.6.1"
+    "lbh-frontend": "^3.6.1",
+    "react-test-renderer": "^17.0.2"
```
Comment (vadim-kirpun) [inline package.json:32]: Do we need this? I think `@testing-library/*` packages are enough

File: src/components/layout/GrayBox/GrayBox.test.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (vadim-kirpun) [inline src/components/layout/GrayBox/GrayBox.test.tsx:1]: React 17 allows us to not import React for jsx anymore
Let's remove it and amend eslint and tsconfig files to allow this

Can be useful for typescript - https://stackoverflow.com/a/65539274
Rule for eslint can be disabled

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

**Comment by vadim-kirpun** [comment]
Please add `coverage` folder to `.gitignore`
Let's also exclude `.stories.tsx` files from tests so they are not included in the coverage report

File: jest.config.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const customJestConfig = {
+    setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
+    testEnvironment: 'jest-environment-jsdom',
+    coverageReporters: ['json-summary', 'text', 'lcov'],
+    collectCoverageFrom: ['src/**/*.{ts,tsx}'],
```
Comment (vadim-kirpun) [inline jest.config.js:5]: Let's exclude `.stories.tsx` files from here so they are not included in the coverage report

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

File: src/components/layout/GrayBox/GrayBox.test.tsx (hunk 6 above)
Comment (winston0410) [reply src/components/layout/GrayBox/GrayBox.test.tsx:1]: Applying the fix from that SO will gives React is undefined. More manual fix is needed.

Comment (vadim-kirpun) [reply src/components/layout/GrayBox/GrayBox.test.tsx:1]: What do you mean `undefined`? I tried to add `jsx: 'react-jsx'` and TS error is gone

Comment (vadim-kirpun) [reply src/components/layout/GrayBox/GrayBox.test.tsx:1]: If needed React can be imported like this `import * as React from 'react';` which is a more recommended way, so I think there should be no problem if you just add `jsx: 'react-jsx'`

Comment (winston0410) [reply src/components/layout/GrayBox/GrayBox.test.tsx]: Please see my latest commit, I have fixed that.

**Review by vadim-kirpun — APPROVED** [review]

**Pull request merged**